process/accounts/accounts_processor.go

```diff
@@ -100,26 +87,33 @@ func (ap *accountsProcessor) addAddressIfInSelfShard(addresses map[string]struct
 	}
 }
 
-func (ap *accountsProcessor) processAccount(address string) (*schema.AccountBalanceUpdate, error) {
-	//TODO: This only works as long as covalent indexer is part of elrond node binary.
-	// This needs to be changed, so that account content is given as an input parameter, not loaded.
-	pubKey, err := ap.pubKeyConverter.Decode(address)
-	if err != nil {
-		return nil, err
+func (ap *accountsProcessor) processAccount(
+	address string,
+	alteredAccounts map[string]*indexer.AlteredAccount,
+) (*schema.AccountBalanceUpdate, error) {
+	if alteredAccounts == nil {
+		return nil, covalent.ErrNilAlteredAccounts
 	}
 
-	acc, err := ap.accounts.LoadAccount(pubKey)
-	if err != nil {
-		return nil, err
+	acc, ok := alteredAccounts[address]
+	if !ok {
+		return nil, fmt.Errorf("%w while extracting %s from altered accounts", covalent.ErrAccountNotFound, address)
 	}
 
-	account, castOk := acc.(data.UserAccountHandler)
-	if !castOk {
-		return nil, covalent.ErrCannotCastAccountHandlerToUserAccount
+	userBalance := acc.Balance
+	if len(userBalance) == 0 {
```

Comment:
Could we add a tiny test for this branch?

Reply:
added